### ISSUE TITLE
Dependabot/Auth Problem

### ISSUE THREAD

**Issue by donnyv12** [google-github-actions/auth#51]
### TL;DR

When Dependabot triggers a workflow that has the `auth` action, it receives the following error and fails: 
```sh
Error: Action failed with error: Error: Error message: Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable
```
When an authenticated user re-runs the workflow, the auth action is successful. 

permissions in the workflow: 
```yaml
permissions:
  contents: 'read'
  id-token: 'write'
```

### Expected behavior

Auth action should be able to find the env var and succeed

### Observed behavior

Auth action is failing. 

### Action YAML

```yaml
# Configure Authenticate to Google Cloud using OIDC and generate an access token.
        - id: 'auth'
          name: 'Authenticate to Google Cloud'
          uses: 'google-github-actions/auth@v0.4.0'
          with:
            create_credentials_file: true
            activate_credentials_file: true
            workload_identity_provider: redacted
            service_account: redacted
```

### Additional information

NOTE: I am aware this is likely a problem with Dependabot that requires a change on their end to resolve the issue, or is a permissions issue that I need to fix in the auth yaml. I would greatly appreciate any help in pointing me in the right direction to resolving this, even though I am fairly certain this isn't a bug in the auth action. Thank you in advance for any help in resolving this.

**Comment by bharathkkb** [comment]
I believe this is the same as access for secrets when dependabot opens PRs where you need to manually rerun after confirming no malicious updates.
I found some info here 
https://github.com/dependabot/dependabot-core/issues/3253#issuecomment-797125425
and
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions#manually-re-running-a-workflow

**Comment by sethvargo** [comment]
Yup, @bharathkkb is correct. Dependabot does not have access to secrets (which includes OIDC tokens).

**Comment by sethvargo** [comment]
I'm going to close this out, since this is working as GitHub intended for secrets ask linked above.
